sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -221,7 +224,8 @@ def _create_pipeline(self, credential, **kwargs):
         # type: (Any, **Any) -> Tuple[Configuration, Pipeline]
         self._credential_policy = None
         if hasattr(credential, "get_token"):
-            self._credential_policy = StorageBearerTokenCredentialPolicy(credential)
+            audience = kwargs.pop('audience', None)
+            self._credential_policy = StorageBearerTokenCredentialPolicy(credential=credential, audience=audience)
```

Comment:
In my current implementation, we are still passing along the Audience object. Do you think it would make more sense to just parse out the value here and then pass it along as a string?